### DISCUSSION TITLE
I tried use sidebar with data table inside

### DISCUSSION THREAD

**Discussion by miguedev1047** [shadcn-ui/ui#5582] [Q&A] [answered]
Hello, I don't know if it's my mistake or the sidebar component's error, but when I try to use the <DataTable/> component within a <SidebarProvider/> its size does not adjust well, causing a horizontal scroll to appear at the bottom

![Sidebar_image](https://github.com/user-attachments/assets/ac6039ff-e9ef-41c6-8ea0-ee50aa65f048)

https://github.com/user-attachments/assets/1c30f686-c4b0-4995-a7f4-1282be3cc4f3

Here I attach a video with my project, I have the same problem with the horizontal scroll

`
 <SidebarProvider>
      <PanelSidebar />     
       ...
      <main className='w-full'>{children}</main>
    </SidebarProvider>
`
And this is the structure that I manage

**Comment by henriquemtn** [comment]
Hi, In my case works only using `overflow-hidden` inside of main content className.

**Comment by beysach** [comment, marked as answer]
```return (
  <SidebarProvider>
    <AppSidebar />
    <SidebarInset className="overflow-hidden">
      <Header />
      <div className="flex-1 flex flex-col mx-auto p-6 w-full">{children}</div>
    </SidebarInset>
  </SidebarProvider>
);```

**Reply by FrancisCodex** [reply]
I was debugging for 8 hours because of this problem 🤣 and it was solved with just 1 classname, Thanks ser 🫡

**Reply by beysach** [reply]
You're welcome hehe ☺️

**Reply by zeshhaan** [reply]
wow man, just wow! came here after debugging manually for almost a whole day. after 3-4 tries, windsurf fixed it for me, but came here and saw it was the same solution.

out of curiosity, why would we need to do this? will there be any consequence that i should be aware of?

**Reply by miguedev1047** [reply]
I did something more complex from the sidebar component that worked for me anyway, but this solution is simpler

**Reply by Ousbaiy** [reply]
it was frustrating debugging this issue all day. thank you

**Reply by emretfn** [reply]
When we do this, we cannot use sticky header. I don't want to use fixed, I will use other elements on the page as sticky. Do you have a solution for this?

**Reply by kazi331** [reply]
> ```
>   <SidebarProvider>
>     <AppSidebar />
>     <SidebarInset className="overflow-hidden">
>       <Header />
>       <div className="flex-1 flex flex-col mx-auto p-6 w-full">{children}</div>
>     </SidebarInset>
>   </SidebarProvider>
> );```
> ```

Thank you so much bro. It saves my hours to debug.

**Reply by Emirhanyld** [reply]
> When we do this, we cannot use sticky header. I don't want to use fixed, I will use other elements on the page as sticky. Do you have a solution for this?

A little late but if someone is still looking for a solution you can use "contain-inline-size" instead of "overflow-hidden". It worked on my project.

**Reply by HIYJOKOU** [reply]
you are god, i've wasted so much time debugging this. tysm

**Reply by Cha0yue** [reply]
> 
> A little late but if someone is still looking for a solution you can use "contain-inline-size" instead of "overflow-hidden". It worked on my project.

Holy moly it works. I wonder if this is a problem with shadcn sidebar itself. Even if I tried using a table made with html tags instead of tanstack table, the problem still occurs. But everything is fixed with just one simple utility class.
That said, can't seem to find that class in tailwindcss docs.

**Reply by ethandpowers** [reply]
Actual godsend!  Not gonna lie, I had to google that one.

**Reply by turnerlewis** [reply]
> > When we do this, we cannot use sticky header. I don't want to use fixed, I will use other elements on the page as sticky. Do you have a solution for this?
> 
> A little late but if someone is still looking for a solution you can use "contain-inline-size" instead of "overflow-hidden". It worked on my project.

Not sure if necroing is a thing but I really don't care. 

Thank you so much Emirhanyld for the solution! Hours of frustration solved 😄

**Reply by Maadtin** [reply]
This is not working tho, scroll appears around main tag but not around table wrapper.

**Comment by FaiJai** [comment]
![image](https://github.com/user-attachments/assets/c63d7ffa-890c-44da-bbea-c154a89fd8de)

Try to add a ```overflow-x-hidden``` to the pink div. If you want it able to scroll vertically as well, use ```overflow-x-hidden overflow-y-auto```
